File: MD.CMS.BusinessLogic.Core/DataAccess/Entities/ContentTypeDefinitionFolderDataBoundSync.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace MD.CMS.BusinessLogic.Core.DataAccess.Entities
{
    public class ContentTypeDefinitionFolderDataBoundSync
    {
        #region Enum
        public enum ContentTypeDefinitionFolderDataBoundSyncType
        {
            NoSync = 0,
            RemoteToOmega = 1,
            OmegaToRemote = 2,
            Bidirectional = 3
        }
        #endregion

        #region Attributes
        private long _folderId;
        private long _contentTypeDefinitionId;
        private DateTime _startTime;
        private DateTime? _endTime;
        private TimeSpan _frequency;
        private bool _enabled;
        private ContentTypeDefinitionFolderDataBoundSyncType _syncType;
        private long? _deltaFieldId;
        #endregion

        #region Properties
        public long FolderId { get => _folderId; set => _folderId = value; }
        public long ContentTypeDefinitionId { get => _contentTypeDefinitionId; set => _contentTypeDefinitionId = value; }
        public DateTime StartTime { get => _startTime; set => _startTime = value; }
        public DateTime? EndTime { get => _endTime; set => _endTime = value; }
        public TimeSpan Frequency { get => _frequency; set => _frequency = value; }
        public bool Enabled { get => _enabled; set => _enabled = value; }
        public ContentTypeDefinitionFolderDataBoundSyncType SyncType { get => _syncType; set => _syncType = value; }
        public long? DeltaFieldId { get => _deltaFieldId; set => _deltaFieldId = value; }
        #endregion
    }
}

File: MD.CMS.Administration/MD.CMS.Administration.Core.Web/scripts/businessLogic/dataAccess/entities/contentTypeDefinitionFolderDataBoundSync.ts
/// <reference path="./base/iBaseEntity.ts" />
/// <reference path="./base/BaseEntity.ts" />
/// <reference path="./base/BaseEntity.ts" />
namespace mdBusinessLogic.dataAccess.entities {

    export class contentTypeDefinitionFolderDataBoundSync implements base.IBaseEntity<contentTypeDefinitionFolderDataBoundSync>{
        public FolderId: number;
        public ContentTypeDefinitionId: number;
        public StartTime: Date;
        public EndTime: Date;
        public Frequency: number;
        public SyncType: contentTypeDefinitionFolderDataBoundSyncType;
        public DeltaFieldId: number;

        constructor(obj?: contentTypeDefinitionFolderDataBoundSync) {
            this.FolderId = 0;
            this.ContentTypeDefinitionId = 0;
            this.StartTime = new Date();
            this.EndTime = null;
            this.Frequency = (60 * 60 * 12);
            this.SyncType = contentTypeDefinitionFolderDataBoundSyncType.NoSync;
            this.DeltaFieldId = null;
            if (obj != null) {
                this.construct(obj);
            }
        }

        public construct(data: any) {
            this.FolderId = helpers.entityHelper.getValue<number>(data, 'FolderId', 0);
            this.ContentTypeDefinitionId = helpers.entityHelper.getValue<number>(data, 'ContentTypeDefinitionId', 0);
            this.StartTime = helpers.entityHelper.getValue<Date>(data, 'StartTime', new Date());
            this.EndTime = helpers.entityHelper.getValue<Date>(data, 'EndTime', null);
            this.Frequency = helpers.entityHelper.getValue<number>(data, 'Frequency', (60 * 60 * 12));
            this.SyncType = helpers.entityHelper.getValue<contentTypeDefinitionFolderDataBoundSyncType>(data, 'SyncType', contentTypeDefinitionFolderDataBoundSyncType.NoSync);
            this.DeltaFieldId = helpers.entityHelper.getValue<number>(data, 'DeltaFieldId', null);
        }

        public clone(): contentTypeDefinitionFolderDataBoundSync {
            return new contentTypeDefinitionFolderDataBoundSync(this);
        }
    }

    export enum contentTypeDefinitionFolderDataBoundSyncType {
        NoSync = 0,
        RemoteToOmega = 1,
        OmegaToRemote = 2,
        Bidirectional = 3
    }
}
